python/ray/data/grouped_dataset.py

```diff
@@ -144,6 +146,111 @@ def _aggregate_on(
         )
         return self.aggregate(*aggs)
 
+    def map_groups(
+        self,
+        fn: Union[CallableClass, Callable[[BatchType], BatchType]],
+        *,
+        compute: Optional[str] = None,
+        batch_format: str = "native",
+        **ray_remote_args,
+    ) -> "Dataset[Any]":
+        """Apply the given function to each group of records of this dataset.
+
+        While the map_groups() is very flexible, note that it comes with downsides:
+            - It may be slower than using more specific methods such as min(), max().
+            - It requires that each group fits in memory on a single node.
+        In general, prefer to use aggregate() methods instead of map_groups().
+
+        This is a blocking operation.
+
+        Examples:
+            >>> # Return a single record per group (list of multiple records in,
+            >>> # list of a single record out).
+            >>> ray.data.range(100).groupby(lambda x: x % 3).map_groups(
+            ...     lambda x: [min(x) / max(x)])
```

Comment:
You'd normally just use aggregate() for this, so how about changing this to a holistic function like median? Can say median is not distributive, so cannot use aggregate().

Reply:
Great point!Done.